### ISSUE TITLE
Closure Compiler errors are being ignored

### ISSUE THREAD

**Issue by osener** [emezeske/lein-cljsbuild#432]
Sample output:

```
$ lein cljsbuild once prod
Compiling ClojureScript...
Compiling "resources/public/js/main.js" from ["src/cljs"]...
Dec 15, 2015 10:11:52 PM com.google.javascript.jscomp.LoggerErrorManager println
SEVERE: /home/ozan/vcs/foo/target/cljsbuild-compiler-1/foo/general.js:255: ERROR - Parse error. Duplicate parameter name "on_click__$1"
return cljs.core.apply.cljs$core$IFn$_invoke$arity$3(React.DOM.div,{"className": "general", "onClick": om_tools.dom.format_opts(on_click__$1)},cljs.core.flatten((new cljs.core.PersistentVector(null,1,(5),cljs.core.PersistentVector.EMPTY_NODE,[cljs.core.apply.cljs$core$IFn$_invoke$arity$3(React.DOM.div,{"className": "right"},cljs.core.flatten((new cljs.core.PersistentVector(null,2,(5),cljs.core.PersistentVector.EMPTY_NODE,[cljs.core.apply.cljs$core$IFn$_invoke$arity$3(React.DOM.div,{"className": "text"},cljs.core.flatten((new cljs.core.PersistentVector(null,1,(5),cljs.core.PersistentVector.EMPTY_NODE,[self__.body],null)))),(function (){var G__55924 = {"className": "dismiss-button icon icon-close-x", "onClick": om_tools.dom.format_opts(((function (on_click__$1,___$1,map__55919,map__55919__$1,body__$1,notification__$1,on_click__$1,poster__$1,tooltip_html__$1,owner__$2,G__55916,map__55917,map__55917__$1,body,notification,on_click,poster,tooltip_html,owner,validate__38309__auto__,ufv___55928,output_schema55899_55929,input_schema55900_55930,input_checker55901_55931,output_checker55902_55932){
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   ^

Dec 15, 2015 10:11:52 PM com.google.javascript.jscomp.LoggerErrorManager printSummary
WARNING: 1 error(s), 0 warning(s)
ERROR: JSC_DUPLICATE_PARAM. Parse error. Duplicate parameter name "on_click__$1" at /home/ozan/vcs/foo/target/cljsbuild-compiler-1/foo/general.js line 255 : 755
Successfully compiled "resources/public/js/main.js" in 40.279 seconds.
```

It is hard to provide a minimal testcase for it as I'm hitting [this bug](https://github.com/Prismatic/om-tools/issues/81) which causes ClojureScript compiler to emit broken code. The bug itself is easy to work around but it has caused us trouble multiple times because the build is marked as successful and goes through our build system with a broken JS output that only contains the files listed in `:preamble`.

Using `org.clojure/clojurescript "1.7.170"` and `lein-cljsbuild "1.1.1"`

**Comment by ikitommi** [comment]
got bitten by this too.

**Comment by danielcompton** [comment]
@osener if you run the bare ClojureScript compiler do you still get a "Successfully compiled" message?" or equivalent `0` exit code? It looks like this may be a ClojureScript compiler bug possibly?

**Comment by mneise** [comment]
Yes, @danielcompton is right. I am not sure if I would call it a bug, but ClojureScript currently doesn't throw an error when there is an error during optimization, but only prints the error to stdout. So there is no way for lein-cljsbuild to know that optimization has failed.

**Comment by danielcompton** [comment]
@osener this might be helpful http://jakemccrary.com/blog/2015/12/19/clojurescript-treat-warnings-as-errors/

**Comment by osener** [comment]
@danielcompton thanks for the link. I just gave it a try but unfortunately compiler doesn't seem to call warning handlers for optimization errors.

**Comment by timgilbert** [comment]
Is there an upstream (CLJS-XXX) ticket for this anywhere? I took a look but didn't see one; the closest I could find was [CLJS-1159](http://dev.clojure.org/jira/browse/CLJS-1159) but there's not a lot of data there.

FWIW, I was able to reproduce this bug by having some files in my classpath that were using `kebab-case.cljs` rather than `snake_case.cljs`; I was invoking the compiler via `lein cljsbuild once dev min` (so running two builds at once). The `dev` build (`:optimizations :none`) succeeded, which I'm guessing is itself a bug, but the `min` build (`:optimizations :advanced`) failed with errors similar to this:

```
Oct 14, 2016 6:50:56 PM com.google.javascript.jscomp.LoggerErrorManager printSummary
WARNING: 3 error(s), 0 warning(s)
ERROR: JSC_MISSING_PROVIDE_ERROR. required "xyzzy.views.set_password" namespace never provided at /home/ubuntu/xyzzy/resources/public/js/compiled/min/out/xyzzy/routes.js line 4 : 0
[...]
Successfully compiled "resources/public/js/compiled/min/main.js" in 41.41 seconds.
```

The overall `lein cljsbuild` process still returned 0, however, and the compiler printed a success error message.

**Comment by lsenta** [comment]
I have the same issue, SEVERE and ERRORS ends with a successful message, no error, and what appears to be a corrupted output.js.

Looks like you can reproduce this bug with a simple javascript files that uses some ES6 features
https://github.com/lsenta/awkward-majine/blob/5ee093ed270d6d0acfcdb6dc85fccb7e8c8585aa/src/cljs/awkward_majine/buffers.js

```
laurent at HALIX in ~/dev/singulargarden/awkward-majine on master with unpushed
› lein clean && lein cljsbuild once min
Compiling ClojureScript...
Compiling "resources/public/js/compiled/app.js" from ["src/cljs"]...
Jul 27, 2017 2:44:20 PM com.google.javascript.jscomp.LoggerErrorManager println
SEVERE: /Users/laurent/dev/singulargarden/awkward-majine/src/cljs/awkward_majine/buffers.js:7: ERROR - this language feature is only supported in es6 mode: const declaration. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features.
  const numberOfChannels = buffers.map(x => x.numberOfChannels)
  ^

Jul 27, 2017 2:44:20 PM com.google.javascript.jscomp.LoggerErrorManager println
SEVERE: /Users/laurent/dev/singulargarden/awkward-majine/src/cljs/awkward_majine/buffers.js:7: ERROR - this language feature is only supported in es6 mode: arrow function. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features.
  const numberOfChannels = buffers.map(x => x.numberOfChannels)
...
ERROR: ES6_FEATURE. this language feature is only supported in es6 mode: const declaration. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features. at /Users/laurent/dev/singulargarden/awkward-majine/src/cljs/awkward_majine/buffers.js line 42 : 4
ERROR: ES6_FEATURE. this language feature is only supported in es6 mode: const declaration. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features. at /Users/laurent/dev/singulargarden/awkward-majine/src/cljs/awkward_majine/buffers.js line 44 : 4
ERROR: ES6_FEATURE. this language feature is only supported in es6 mode: arrow function. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features. at /Users/laurent/dev/singulargarden/awkward-majine/src/cljs/awkward_majine/buffers.js line 44 : 33
ERROR: ES6_FEATURE. this language feature is only supported in es6 mode: const declaration. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features. at /Users/laurent/dev/singulargarden/awkward-majine/src/cljs/awkward_majine/buffers.js line 53 : 2
ERROR: ES6_FEATURE. this language feature is only supported in es6 mode: const declaration. Use --language_in=ECMASCRIPT6 or ECMASCRIPT6_STRICT to enable ES6 features. at /Users/laurent/dev/singulargarden/awkward-majine/src/cljs/awkward_majine/buffers.js line 57 : 2
Successfully compiled "resources/public/js/compiled/app.js" in 17.674 seconds.
```

**Comment by acron0** [comment]
I was bitten by the bug today...

**Comment by ChillkroeteTTS** [comment]
Me too... several times now. I have a react native project and it's really unfortunate if your automated build process uploads a broken artifact to the app store.

**Comment by mneise** [comment]
This has been fixed in ClojureScript with the release of [1.10.238](https://github.com/clojure/clojurescript/releases/tag/r1.10.238). The ticket for this issue is [CLJS-2495](https://dev.clojure.org/jira/browse/CLJS-2495).
